### PR TITLE
[Doc] Add two parameters to limit the tenant count per cluster and namespace count per tenant

### PR DESCRIPTION
**Pull request by momo-jun**
### Modifications

To align with https://github.com/apache/pulsar/pull/8261 and https://github.com/apache/pulsar/pull/8267, add the following two parameters and explanations to the broker configurations.
- 'maxTenants'
- 'maxNamespacePerTenant'

> Note: This change will be applied to more historical versions since 2.7.0 as soon as the doc PR gets approval.

### Documentation
  
- [x] `doc`

### CODE REVIEW

**Comment by momo-jun** [comment]
Hi @315157973 @codelipenghui @Anonymitaet, can you pls review this doc PR? Thanks.
Please kindly note that the same change will be applied to more historical versions since 2.7.0 as soon as this doc PR gets approval.

File: site2/docs/reference-configuration.md [hunk 1]
```diff
@@ -164,6 +164,7 @@ Pulsar brokers are responsible for handling incoming messages from producers, di
 |bindAddresses| Additional Hostname or IP addresses the service binds on: `listener_name:scheme://host:port,...`.  ||
 |advertisedAddress| Hostname or IP address the service advertises to the outside world. If not set, the value of `InetAddress.getLocalHost().getHostName()` is used.  ||
 |clusterName| Name of the cluster to which this broker belongs to ||
+|maxTenants|The maximum number of tenants that can be created in each pulsar cluster. When the number of tenants reaches the threshold, the broker rejects the request of creating a new tenant. The default value 0 disables the check. |0|
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:167]:
```suggestion
|maxTenants|The maximum number of tenants that can be created in each Pulsar cluster. When the number of tenants reaches the threshold, the broker rejects the request of creating a new tenant. The default value 0 disables the check. |0|
```

File: site2/docs/reference-configuration.md [hunk 2]
```diff
@@ -363,6 +364,7 @@ brokerServiceCompactionThresholdInBytes|If the estimated backlog size is greater
 | preciseTopicPublishRateLimiterEnable | Enable precise topic publish rate limiting. | false |
 | lazyCursorRecovery | Whether to recover cursors lazily when trying to recover a managed ledger backing a persistent topic. It can improve write availability of topics. The caveat is now when recovered ledger is ready to write we're not sure if all old consumers' last mark delete position(ack position) can be recovered or not. So user can make the trade off or have custom logic in application to checkpoint consumer state.| false |  
 |haProxyProtocolEnabled | Enable or disable the [HAProxy](http://www.haproxy.org/) protocol. |false|
+| maxNamespacesPerTenant | The maximum number of namespaces that can be created in each tenant. When the number of namespaces reaches this threshold, the broker rejects the request of creating a new tenant. The default value 0 disables the check. |0|
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:367]:
```suggestion
| maxNamespacesPerTenant | The maximum number of namespaces that can be created in each tenant. When the number of namespaces reaches this threshold, the broker rejects the request of creating a new tenant. The default value 0 disables the check. |0|
```

**Review by Anonymitaet — DISMISSED** [review]

**Review by Anonymitaet — APPROVED** [review]

**Comment by momo-jun** [comment]
Hi @Anonymitaet, the changes have been applied to more historical versions since 2.7.0 and the PR can be merged. Thanks.

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**